subgraphs/harvest-finance/src/vault.ts

```diff
@@ -147,6 +145,33 @@ function handleMint(event: TransferEvent): void {
 
   vault.outputTokenSupply = vault.outputTokenSupply!.plus(amount)
 
+  vault.outputTokenPriceUSD = vault.totalValueLockedUSD.div(
+    decimals.fromBigInt(vault.outputTokenSupply!, 6)
+  )
+
+  vault.save()
+
+  metrics.updateFinancials(event.block)
+  metrics.updateUsageMetrics(event.block, event.transaction.from)
+  metrics.updateVaultSnapshots(vaultAddress, event.block)
+}
+
+function handleBurn(event: TransferEvent): void {
+  const amount = event.params.value
+  const vaultAddress = event.address
+
+  const vault = Vault.load(vaultAddress.toHexString())
+
+  if (!vault) return
+
+  const outputTokenSupply = vault.outputTokenSupply! || BigInt.fromI32(0)
```

Comment:
Probably just using vault.outputTokenSupply is enough because its value is initialized always to zero upon vault creation in vaults.initialize

Reply:
we are doing Vault.load here. the reason why I did that is because the way the assembly script compiler works. anyways I will be more clear using a simple if statement as we did in the mint function :)